integration/update_test.go

```diff
@@ -279,3 +279,186 @@ func TestUpdateIncOperator(t *testing.T) {
+		"OkSetString": {
+			id:     "string",
+			update: bson.D{{"$set", bson.D{{"value", "ok value"}}}},
+			result: bson.D{{"_id", "string"}, {"value", "ok value"}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+		},
+		"ArrayNil": {
+			id:     "string",
+			update: bson.D{{"$set", bson.D{{"value", bson.A{nil}}}}},
+			result: bson.D{{"_id", "string"}, {"value", bson.A{nil}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+		},
+		"FieldNotExist": {
+			id:     "string",
+			update: bson.D{{"$set", bson.D{{"foo", int32(1)}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			result: bson.D{{"_id", "string"}, {"value", "foo"}, {"foo", int32(1)}},
+		},
+		"Double": {
+			id:     "double",
+			update: bson.D{{"$set", bson.D{{"value", float64(1)}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			result: bson.D{{"_id", "double"}, {"value", float64(1)}},
+		},
+		"NaN": {
+			id:     "double",
+			update: bson.D{{"$set", bson.D{{"value", math.NaN()}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			result: bson.D{{"_id", "double"}, {"value", math.NaN()}},
+		},
+		"EmptyArray": {
+			id:     "double",
+			update: bson.D{{"$set", bson.D{{"value", bson.A{}}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			result: bson.D{{"_id", "double"}, {"value", bson.A{}}},
+		},
+		"Null": {
+			id:     "double",
+			update: bson.D{{"$set", bson.D{{"value", nil}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			result: bson.D{{"_id", "double"}, {"value", nil}},
+		},
+		"Int32": {
+			id:     "double",
+			update: bson.D{{"$set", bson.D{{"value", int32(1)}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			result: bson.D{{"_id", "double"}, {"value", int32(1)}},
+		},
+		"Inf": {
+			id:     "double",
+			update: bson.D{{"$set", bson.D{{"value", math.Inf(+1)}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			result: bson.D{{"_id", "double"}, {"value", math.Inf(+1)}},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ctx, collection := setup(t)
+			_, err := collection.InsertMany(ctx, []any{
+				bson.D{{"_id", "string"}, {"value", "foo"}},
+				bson.D{{"_id", "double"}, {"value", float64(0.0)}},
+			})
+			require.NoError(t, err)
+
+			res, err := collection.UpdateOne(ctx, bson.D{{"_id", tc.id}}, tc.update)
+			if tc.err != nil {
+				if !AssertEqualAltWriteError(t, tc.err, tc.alt, err) {
+					t.Logf("%[1]T %[1]v", err)
+					t.FailNow()
+				}
+				return
```

Comment:
Let's make this block the same as in other tests

Reply:
log removed, added check for res.